### PR TITLE
[TASK-1] Shoe Object Modelling New

### PR DESCRIPTION
**Pull request by SaraiNina**

### CODE REVIEW

File: src/main/java/com/nina/modelling/shoes/ShoeFactory.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.nina.modelling.shoes;
+
+
+public class ShoeFactory extends Shoe {
+
+    public static void main(String[] args) {
+
+        KidShoe kidShoe = new KidShoe("Summer", "Flower", "Orthopedic");
+        KidShoe kidShoe1 = new KidShoe("Summer");
```
Comment (alexandru-corghencea) [inline src/main/java/com/nina/modelling/shoes/ShoeFactory.java]: Avoid using numbers in variable names.
Name it johnsShoe or marysShoe or shoeOnTheWindow

**Review by alexandru-corghencea — CHANGES_REQUESTED** [review]

File: src/main/java/com/nina/modelling/shoes/ShoeFactory.java (hunk 1 above)
Comment (SaraiNina) [reply src/main/java/com/nina/modelling/shoes/ShoeFactory.java:9]: working on it

File: src/main/java/com/nina/modelling/shoes/KidShoe.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private String accessory;
+    private String type;
+
+    public KidShoe(String category, String accessory, String type) {
+        this.category = category;
+        this.accessory = accessory;
+        this.type = type;
+    }
+
+    public KidShoe(String category) {
```
Comment (alexandru-corghencea) [inline src/main/java/com/nina/modelling/shoes/KidShoe.java:8]: contructors with less params should be placed first

File: src/main/java/com/nina/modelling/shoes/ShoeFactory.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.nina.modelling.shoes;
+
+
+public class ShoeFactory extends Shoe {
+
+    public static void main(String[] args) {
+
+        KidShoe kidShoe = new KidShoe("Summer", "Flower", "Orthopedic");
```
Comment (alexandru-corghencea) [inline src/main/java/com/nina/modelling/shoes/ShoeFactory.java]: can be didShoeOrthopedic

File: src/main/java/com/nina/modelling/shoes/WomanShoe.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.nina.modelling.shoes;
+
+public class WomanShoe extends Shoe {
+    private String collections;  //TODO: probably list of strings
```
Comment (alexandru-corghencea) [inline src/main/java/com/nina/modelling/shoes/WomanShoe.java]: probably collections is an list of strings?
maybe accessories also if a shoe has more accessories

File: src/main/resources/Task 1 - Shoe Object Modelling.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Object Modeling Shoes:
+- create a class diagram for shoes object hierarchy.
```
Comment (alexandru-corghencea) [inline src/main/resources/Task 1 - Shoe Object Modelling.txt:2]: on diagram if kidShoe extends Shoe than the arrow should point different direction

File: src/main/resources/Task 1 - Shoe Object Modelling.txt [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+- implement class diagram in Java code.
+- use static code analysis tools to verify code for issues and potential bugs.
+- create a GitHub branch and Pull Request and fix all threads from code review.
```
Comment (alexandru-corghencea) [inline src/main/resources/Task 1 - Shoe Object Modelling.txt:5]: on diagram arrange shoe objects as a vertical tree, like Christmas tree instead spread around the page.

File: src/main/resources/Task 1 - Shoe Object Modelling.txt [hunk 7]
```diff
@@ -0,0 +1,5 @@
+Object Modeling Shoes:
+- create a class diagram for shoes object hierarchy.
+- implement class diagram in Java code.
+- use static code analysis tools to verify code for issues and potential bugs.
```
Comment (alexandru-corghencea) [inline src/main/resources/Task 1 - Shoe Object Modelling.txt:4]: on diagram you are suppose to show objects blueprints: properties and methods. Constructors are not member of objects there fore should not be displayed on diagram.

Comment (alexandru-corghencea) [inline src/main/resources/Task 1 - Shoe Object Modelling.txt:4]: On diagram mark public fields and methods with + and private ones with -

File: src/main/java/com/nina/modelling/shoes/KidShoe.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
+package com.nina.modelling.shoes;
+
+public class KidShoe extends Shoe {
```
Comment (alexandru-corghencea) [inline src/main/java/com/nina/modelling/shoes/KidShoe.java:3]: Please answer for each of the trheads with what was changed, but dont mark as resolved.
I will mark resolve if I agree with them else i will reopen for fix

**Review by alexandru-corghencea — CHANGES_REQUESTED** [review]

File: src/main/java/com/nina/modelling/shoes/ShoeFactory.java (hunk 1 above)
Comment (SaraiNina) [reply src/main/java/com/nina/modelling/shoes/ShoeFactory.java]: Changes the name of variables

File: src/main/java/com/nina/modelling/shoes/KidShoe.java (hunk 2 above)
Comment (SaraiNina) [reply src/main/java/com/nina/modelling/shoes/KidShoe.java:8]: Now it"s first

File: src/main/java/com/nina/modelling/shoes/ShoeFactory.java (hunk 3 above)
Comment (SaraiNina) [reply src/main/java/com/nina/modelling/shoes/ShoeFactory.java]: What do you mean? The name of variable must be name didShoeOrthopedic?

File: src/main/java/com/nina/modelling/shoes/WomanShoe.java (hunk 4 above)
Comment (SaraiNina) [reply src/main/java/com/nina/modelling/shoes/WomanShoe.java]: Collections I put in singular and accessories too. I hope to be ok.

Comment (SaraiNina) [reply src/main/java/com/nina/modelling/shoes/WomanShoe.java]: Named the collections to group, it's synonymy.

File: src/main/java/com/nina/modelling/shoes/ShoeFactory.java (hunk 1 above)
Comment (alexandru-corghencea) [reply src/main/java/com/nina/modelling/shoes/ShoeFactory.java]: not all variables fixed

Comment (SaraiNina) [reply src/main/java/com/nina/modelling/shoes/ShoeFactory.java]: All names of variables was changed.

File: src/main/resources/Task 1 - Shoe Object Modelling.txt (hunk 6 above)
Comment (SaraiNina) [reply src/main/resources/Task 1 - Shoe Object Modelling.txt:5]: The new diagram was drawn.

File: src/main/resources/Task 1 - Shoe Object Modelling.txt (hunk 7 above)
Comment (SaraiNina) [reply src/main/resources/Task 1 - Shoe Object Modelling.txt:4]: The constructors was deleted.

Comment (SaraiNina) [reply src/main/resources/Task 1 - Shoe Object Modelling.txt:4]: The fields  was marked with corresponding operators.

File: src/main/java/com/nina/modelling/shoes/KidShoe.java (hunk 8 above)
Comment (SaraiNina) [reply src/main/java/com/nina/modelling/shoes/KidShoe.java:3]: Ok

**Review by alexandru-corghencea — APPROVED** [review]

**Pull request merged**